opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py

```diff
@@ -41,24 +46,147 @@
     SpanAttributes.NETWORK_PROTOCOL_VERSION,
     SpanAttributes.SERVER_ADDRESS,
     SpanAttributes.SERVER_PORT,
-    # TODO: Support opt-in for scheme in new semconv
-    # SpanAttributes.URL_SCHEME,
+    SpanAttributes.URL_SCHEME,
 ]
 
+_server_duration_attrs_old = [
+    SpanAttributes.HTTP_METHOD,
+    SpanAttributes.HTTP_HOST,
+    SpanAttributes.HTTP_SCHEME,
+    SpanAttributes.HTTP_STATUS_CODE,
+    SpanAttributes.HTTP_FLAVOR,
+    SpanAttributes.HTTP_SERVER_NAME,
+    SpanAttributes.NET_HOST_NAME,
+    SpanAttributes.NET_HOST_PORT,
+]
+
+_server_duration_attrs_new = [
+    _SPAN_ATTRIBUTES_ERROR_TYPE,
+    SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.HTTP_RESPONSE_STATUS_CODE,
+    SpanAttributes.HTTP_ROUTE,
+    SpanAttributes.NETWORK_PROTOCOL_VERSION,
+    SpanAttributes.URL_SCHEME,
+]
+
+_server_active_requests_count_attrs_old = [
+    SpanAttributes.HTTP_METHOD,
+    SpanAttributes.HTTP_HOST,
+    SpanAttributes.HTTP_SCHEME,
+    SpanAttributes.HTTP_FLAVOR,
+    SpanAttributes.HTTP_SERVER_NAME,
+    SpanAttributes.NET_HOST_NAME,
+    SpanAttributes.NET_HOST_PORT,
+]
+
+_server_active_requests_count_attrs_new = [
+    SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.URL_SCHEME,
+]
+
+_OTEL_SEMCONV_STABILITY_OPT_IN_KEY = "OTEL_SEMCONV_STABILITY_OPT_IN"
+
 
-def _filter_duration_attrs(attrs, sem_conv_opt_in_mode):
+class _OpenTelemetryStabilitySignalType:
+    HTTP = "http"
+
+
+class _OpenTelemetryStabilityMode(Enum):
+    # http - emit the new, stable HTTP and networking conventions ONLY
+    HTTP = "http"
+    # http/dup - emit both the old and the stable HTTP and networking conventions
+    HTTP_DUP = "http/dup"
+    # default - continue emitting old experimental HTTP and networking conventions
+    DEFAULT = "default"
```

Comment:
These modes have priorities, that idea should be represented in the `Enum` itself by using numerical values:

```suggestion
    # http/dup - emit both the old and the stable HTTP and networking conventions
    HTTP_DUP = 0
    # http - emit the new, stable HTTP and networking conventions ONLY
    HTTP = 1
    # default - continue emitting old experimental HTTP and networking conventions
    DEFAULT = 2
```

Reply:
The enum values must be the string values that we read from the env vars, not numbers.

Reply:
You already have those string names in the `enum` names:

```python
from enum import Enum
from ipdb import set_trace
from os import environ


class OptIn(Enum):
    dup = 0
    http = 1
    default = 2


opt_in = sorted(
    [
        getattr(OptIn, element) for element in set(
            environ.get(
                "OPENTELEMETRY_SEMCONV_STABILITY_OPT_IN",
                OptIn.default.name
            ).split(",")
        )
        if element in OptIn.__members__
    ],
    key=lambda x: x.value
)[0]

set_trace()
```

```bash
tigre@hilleman:~/sandbox$ python3 opt_in_example.py 
--Return--
None
> /home/tigre/sandbox/opt_in_example.py(25)<module>()
      6 class OptIn(Enum):
      7     dup = 0
      8     http = 1
      9     default = 2
     10 
     11 
     12 opt_in = sorted(
     13     [
     14         getattr(OptIn, element) for element in set(
     15             environ.get(
     16                 "OPENTELEMETRY_SEMCONV_STABILITY_OPT_IN",
     17                 OptIn.default.name
     18             ).split(",")
     19         )
     20         if element in OptIn.__members__
     21     ],
     22     key=lambda x: x.value
     23 )[0]
     24 
---> 25 set_trace()

ipdb> opt_in
<OptIn.default: 2>
ipdb> opt_in.value
2
ipdb> opt_in.name
'default'

```

Reply:
OptIn.Dup.name does not map to "http/dup"

Reply:
> OptIn.Dup.name does not map to "http/dup"

See [this](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2425/files#r1569487644) :v: